gpkit/solution_array.py

```diff
@@ -69,9 +114,33 @@ def subinto(self, posy):
         else:
             return posy.sub(self["variables"])
 
-    def table(self, tables=("cost", "sweepvariables", "freevariables",
-                            "constants", "sensitivities"),
-              latex=False, **kwargs):
+    def _parse_showvars(self, showvars):
+        showvars_out = set()
+        if showvars:
+            for k in showvars:
+                k, _ = self["variables"].parse_and_index(k)
+                keys = self["variables"].keymap[k]
+                showvars_out.update(keys)
+        return showvars_out
+
+    def summary(self, showvars=()):
+        "Print summary table, showing top sensitivities and no constants"
+        showvars = self._parse_showvars(showvars)
+        out = self.table(showvars, ["cost", "sweepvariables", "freevariables"])
+        constants_in_showvars = showvars.intersection(self["constants"])
+        senss_tables = []
+        if len(self["constants"]) < 7 or constants_in_showvars:
+            senss_tables.append("sensitivities")
+        if len(self["constants"]) >= 7:  # 2 more than the default topN
+            senss_tables.append("topsensitivities")
+        senss_str = self.table(showvars, senss_tables)
+        if senss_str:
+            out += "\n" + senss_str
+        return out
+
+    def table(self, showvars=(),
```

Comment:
make name of `showvars` consistent with the controlpanel kwarg that gives control over which variables to show?